solidity/contracts/core/pool-diamond/facets/GetterFacet.sol

```diff
@@ -410,6 +410,40 @@ contract GetterFacet is GetterFacetInterface {
     return (nextPrice * nextSize) / divisor;
   }
 
+  function getNextShortAveragePriceInt(
+    address indexToken,
+    uint256 nextPrice,
+    int256 sizeDelta,
+    int256 realizedPnl
+  ) public view returns (uint256) {
+    // Load diamond storage
+    LibPoolV1.PoolV1DiamondStorage storage ds = LibPoolV1
+      .poolV1DiamondStorage();
+
+    uint256 shortSize = ds.shortSizeOf[indexToken];
+    uint256 shortAveragePrice = ds.shortAveragePriceOf[indexToken];
+    uint256 priceDelta = shortAveragePrice > nextPrice
+      ? shortAveragePrice - nextPrice
+      : nextPrice - shortAveragePrice;
+    uint256 delta = (shortSize * priceDelta) / shortAveragePrice;
```

Comment:
don't know if the following case can be happened, but if shortAveragePrice = 0, then it would reverted w/o returning any value

Reply:
It is actually checked before the call to this function in `PerpTradeFacet`. However, it is a good practice to handle this scenario. I will push a change for this.

Reply:
I think ` if (shortAveragePrice == 0) return nextPrice;` can be placed in `getNextShortAveragePriceInt` as well wdyt

Reply:
if you think it's ok then please re-resolve conversation. thanks!